integrations/spiral/src/Bootloader/SearchBootloader.php

```diff
@@ -82,144 +95,82 @@ public function boot(ConsoleBootloader $console, Container $container): void
 
         $engineServices = [];
         foreach ($engines as $name => $engineConfig) {
-            $adapterServiceId = 'schranz_search.adapter.' . $name;
-            $engineServiceId = 'schranz_search.engine.' . $name;
-            $schemaLoaderServiceId = 'schranz_search.schema_loader.' . $name;
-            $schemaId = 'schranz_search.schema.' . $name;
+            $adapterServiceId = 'schranz_search.adapter.'.$name;
+            $engineServiceId = 'schranz_search.engine.'.$name;
+            $schemaLoaderServiceId = 'schranz_search.schema_loader.'.$name;
+            $schemaId = 'schranz_search.schema.'.$name;
 
             /** @var string $adapterDsn */
             $adapterDsn = $engineConfig['adapter'];
             $dirs = $engineSchemaDirs[$name] ?? [];
 
-            $container->bindSingleton($adapterServiceId, function (Container $container) use ($adapterDsn) {
-                /** @var AdapterFactory $factory */
-                $factory = $container->get(AdapterFactory::class);
-
-                return $factory->createAdapter($adapterDsn);
-            });
+            $container->bindSingleton($adapterServiceId, $adapter = $adapterFactory->createAdapter($adapterDsn));
```

Comment:
This looks very unexpected, if that would be the case we would not even need `$container->bindSingleton` as this way we not even allow service decoration as `$adapter` is used instead of the service inside the container.

Same for the other cases. I would not go this way as from dependency injection point of view this looks not correctly for me.